nextjs-app/sanity/lib/utils.ts

```diff
@@ -36,16 +36,31 @@ export function linkResolver(link: Link | undefined) {
     case "href":
       return link.href || null;
     case "page":
-      if (link?.page) {
+      if (link?.page && typeof link.page === "string") {
         return `/${link.page}`;
       }
-      return null;
     case "post":
-      if (link?.post) {
+      if (link?.post && typeof link.post === "string") {
         return `/posts/${link.post}`;
       }
-      return null;
     default:
       return null;
   }
 }
+
+type DataAttributeConfig = {
+  id: string;
+  type: string;
+  path: string;
+};
+
+export function createDataAttributeConfig(config: DataAttributeConfig) {
+  return {
+    projectId,
+    dataset,
+    baseUrl: studioUrl,
+    id: config.id,
+    type: config.type,
+    path: config.path,
+  };
+}
```

Comment:
You could use `createDataAttribute`'s `combine` function here to simplify things, also maybe good to have the canonical method for combining attribute data.

```suggestion
type DataAttributeConfig = CreateDataAttributeProps &
  Required<Pick<CreateDataAttributeProps, "id" | "type" | "path">>;

export function dataAttr(config: DataAttributeConfig) {
  return createDataAttribute({
    projectId,
    dataset,
    baseUrl: studioUrl,
  }).combine(config);
}
```